manifests/init.pp

```diff
@@ -27,10 +30,16 @@
   $valid_ensures = [ 'absent', 'present', 'latest' ]
   validate_re($ensure, $valid_ensures)
 
+  file { $name:
```

Comment:
Ah missed that, thanks. So how about we move the new file resource at line 33 inside the else?

Like this:
```puppet
...
} else {
  file { $name:
    owner => $owner,
    group => $group,
    mode  => $mode,
  }

  $real_source = "https://${s3_domain}/${source}"
  ...
}
```